### DISCUSSION TITLE
Enable faster training by pinning data to memory

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#58] [Archived]
<USER>>>> btomtom5<br/> <DATE> [June 21, 2019, 5:55am]<br/> <br/> Is there a reason that the DataLoader sets pin_memory=false?<br/> <br/> loader = DataLoader(<br/> dataset,<br/> batch_size=c.eval_batch_size if is_val else c.batch_size,<br/> shuffle=False,<br/> collate_fn=dataset.collate_fn,<br/> drop_last=False,<br/> sampler=sampler,<br/> num_workers=c.num_val_loader_workers<br/> if is_val else c.num_loader_workers,<br/> pin_memory=False)<br/> return loader<br/> <br/> As I understand it, pin_memory speeds up transfers from host to device.<br/> Perhaps it's because it has been determined that this is not a<br/> bottleneck in the training?<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/enable-faster-training-by-pinning-data-to-memory]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [June 21, 2019, 8:18am]<br/> <br/> This is one of the pytorch core developers.<br/> <br/> > try to use pin_memory, but if you are seeing system freeze or swap<br/> > being used a lot, disable it.<br/> <br/> So this is not the case for my training. But consider it for your own.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> btomtom5<br/> <DATE> [June 21, 2019, 4:47pm]<br/> <br/> ! Makes sense!<br/> <br/> [Archived Post]<br/>
